File: app/src/main/java/com/jossy/android/mobilebenchmarkappjava/BenchmarkApplication.java
package com.jossy.android.mobilebenchmarkappjava;

import static android.os.Process.THREAD_PRIORITY_BACKGROUND;
import static android.os.Process.setThreadPriority;

import android.app.Application;

import com.bumptech.glide.Glide;


public class BenchmarkApplication extends Application {

    public static final String RESULT = "result";

    @Override
    public void onCreate() {
        super.onCreate();
        initializeComponents();
    }

    private void initializeComponents() {
        new Thread(() -> {
            setThreadPriority(THREAD_PRIORITY_BACKGROUND);
            Glide.get(this);
        }).start();
    }

    @Override
    public void onTrimMemory(int level) {
        super.onTrimMemory(level);
        if (level >= TRIM_MEMORY_MODERATE) {
            Glide.get(this).clearMemory();
        }
    }

    @Override
    public void onLowMemory() {
        super.onLowMemory();
        Glide.get(this).clearMemory();
    }
}

File: app/src/main/java/com/jossy/android/mobilebenchmarkappkotlin/BenchmarkApplication.kt
package com.jossy.android.mobilebenchmarkappkotlin

import android.app.Application
import android.os.Process.setThreadPriority
import coil.Coil

class BenchmarkApplication : Application() {
    companion object {
        const val RESULT = "result"
    }

    override fun onCreate() {
        super.onCreate()
        Thread {
            setThreadPriority(android.os.Process.THREAD_PRIORITY_BACKGROUND)
            Coil.imageLoader(this)
        }.start()
    }
}
